@here/harp-mapview/lib/geometry/AddGroundPlane.ts

```diff
@@ -0,0 +1,196 @@
+ */
+
+import { GeometryKind } from "@here/harp-datasource-protocol";
+import {
+    EdgeLengthGeometrySubdivisionModifier,
+    SubdivisionMode
+} from "@here/harp-geometry/lib/EdgeLengthGeometrySubdivisionModifier";
+import { SphericalGeometrySubdivisionModifier } from "@here/harp-geometry/lib/SphericalGeometrySubdivisionModifier";
+import { Projection, ProjectionType } from "@here/harp-geoutils";
+import { MapMeshBasicMaterial, MapMeshStandardMaterial } from "@here/harp-materials";
+import * as THREE from "three";
+
+import { Tile } from "../Tile";
+import { LodMesh } from "./LodMesh";
+import { projectTilePlaneCorners } from "./ProjectTilePlaneCorners";
+import { registerTileObject } from "./RegisterTileObject";
+
+const tmpV = new THREE.Vector3();
+
+/**
+ * Creates and adds a background plane mesh for the tile.
+ * @param tile - The tile to which the ground plane belongs.
+ * @param renderOrder - The plane render order.
+ * @param materialOrColor - The plane material or a color for a default material.
+ * @param createTexCoords - Whether to create texture coordinates.
+ * @param receiveShadow - Whether the plane should receive shadows.
+ * @param createMultiLod - Whether to generate multiple LODs for sphere projection.
+ * @internal
+ */
+export function addGroundPlane(
+    tile: Tile,
+    renderOrder: number,
+    materialOrColor: THREE.Material | THREE.Material[] | number = tile.mapView.clearColor,
+    createTexCoords: boolean = false,
+    receiveShadow: boolean = tile.mapView.shadowsEnabled,
+    createMultiLod: boolean = tile.mapView.enableMixedLod !== false
+) {
+    const mesh = createGroundPlane(
+        tile,
+        createTexCoords,
+        receiveShadow,
+        materialOrColor,
+        createMultiLod
+    );
+    mesh.receiveShadow = receiveShadow;
+    mesh.renderOrder = renderOrder;
+    registerTileObject(tile, mesh, GeometryKind.Background, { pickable: false });
+    tile.objects.push(mesh);
+}
+
+function createGroundPlane(
+    tile: Tile,
+    createTexCoords: boolean,
+    receiveShadow: boolean,
+    materialOrColor: THREE.Material | THREE.Material[] | number,
+    createMultiLod: boolean
+): THREE.Mesh {
+    const { dataSource, projection } = tile;
+    const sourceProjection = dataSource.getTilingScheme().projection;
+    const shouldSubdivide = projection.type === ProjectionType.Spherical;
+    const useLocalTargetCoords = !shouldSubdivide;
+
+    const material =
+        typeof materialOrColor === "number"
+            ? createGroundPlaneMaterial(
+                  new THREE.Color(materialOrColor),
+                  receiveShadow,
+                  projection.type === ProjectionType.Spherical
+              )
+            : materialOrColor;
+
+    const geometry = createGroundPlaneGeometry(
+        tile,
+        useLocalTargetCoords,
+        createTexCoords,
+        receiveShadow
+    );
+
+    if (!shouldSubdivide) {
+        return new THREE.Mesh(geometry, material);
```

Comment:
You need to convert the Float64Array back to Float32Array also in this case.

Reply:
If !shouldSubdivide, then useLocalTargetCoords == true. You can see in createPlaneGeometry() that in that case whe use FLoat32Array:

```
if (useLocalTargetCoords) {
        for (const corner of cornersArray) {
            projection.reprojectPoint(sourceProjection, corner, corner).sub(tile.center);
        }
    }
```

Reply:
I'm not such a fan of having the useLocalTargetCoords control the transformation to local coordinates in the `createPlaneGeometry` function, I think it makes it easier to read when this transformation is explicitly outside of the function, like it was previously with `moveTileCenter`, so that it isn't responsible for too many things. But because @FraukeF already approve, I won't block it.